### PR TITLE
Part of #7272 - Centralize Workflows, Status, and Exception Management

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes:

Part of #7272

Pending:
- Profiler, Data Insights and Data Quality workflows should be created from `BaseWorkflow`
- Status handling needs to be updated for the workflows above after the update
- Clean the legacy Sink, Processor, and BulkSink classes in favor of using `steps.*`
- Re-implement the `test_workflow_output_handler` with the new status structure

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/src/metadata/ingestion/api/topology_runner.py [hunk 1]
```diff
@@ -182,17 +187,21 @@ def sink_request(self, stage: NodeStage, entity_request: C) -> Iterable[Entity]:
         """
 
         # Either use the received request or the acknowledged Entity
-        entity = entity_request
+        entity = entity_request.right
 
         if stage.nullable and entity is None:
```
Comment (TeddyCr) [inline ingestion/src/metadata/ingestion/api/topology_runner.py:192]: wouldn't this be `if not stage.nullable` if something can be nullable (ie `stage.nullable is True`) then we should not throw a ValueError right?

File: ingestion/src/metadata/workflow/base.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        by each ingestion workflow.
+
+        Pass each record from the source down the pipeline:
+        Source -> (Processor) -> Sink
+        or Source -> (Processor) -> Stage -> BulkSink
+
+        Note how the Source class needs to be an Iterator. Specifically,
+        we are defining Sources as Generators.
+        """
+        for record in self.source.run():
```
Comment (TeddyCr) [inline ingestion/src/metadata/workflow/base.py:128]: where do we enforce that `Source` should have a run()

Comment (pmbrull) [reply ingestion/src/metadata/workflow/base.py:128]: [here](https://github.com/open-metadata/OpenMetadata/pull/13029/files#diff-0299422e9540c993001cd05eea601d12500c7f3de9b589d596625458e0346260R144). Might add the `run` as an abstractmethod to the Step class itself

Comment (TeddyCr) [reply ingestion/src/metadata/workflow/base.py:128]: the `Source` class inherit from the `StageStep` class?

**Comment by pmbrull** [comment]
> @pmbrull, could you add a UML diagram of the class relationship somewhere, it would help get a better overview of the structure and the different relationships between objects

I'll take care of it. I'll publish it in another PR

**Review by TeddyCr — APPROVED** [review]

**Pull request merged**